src/pretalx/schedule/forms.py

```diff
@@ -145,10 +206,37 @@ def clean_availabilities(self):
             if isinstance(self.data.get("availabilities"), list)
             else self._parse_availabilities_json(data)
         )
-        availabilities = []
 
+        # validate raw data and parse datetimes
         for rawavail in rawavailabilities:
             self._validate_availability(rawavail)
+
+        # sort availabilities, so we can merge ones that touch or overlap
+        rawavailabilities.sort(key=itemgetter("start"))
+
+        tz = pytz.timezone(self.event.timezone)
+
+        availabilities = []
+        for rawavail in rawavailabilities:
+            if self.availabilities_within:
+                # check if this availability entry is within availabilies_within
+                for availability in self.availabilities_within:
```

Comment:
The `Availability` class already provides methods to find overlaps and inclusions, would probably be easier (less code or duplication) to use those.